apps/meteor/server/lib/moderation/reportMessage.ts

```diff
@@ -0,0 +1,57 @@
+import { Messages, ModerationReports, Rooms, Users } from '@rocket.chat/models';
+import type { IMessage, IUser } from '@rocket.chat/core-typings';
+
+import { canAccessRoomAsync } from '../../../app/authorization/server/functions/canAccessRoom';
+import { AppEvents, Apps } from '../../../ee/server/apps';
+
+export const reportMessage = async (messageId: IMessage['_id'], description: string, uid: IUser['_id']) => {
+	if (!uid) {
+		throw new Error('error-invalid-user');
+	}
+
+	if (!description.trim()) {
+		throw new Error('error-invalid-description');
+	}
+
+	const message = await Messages.findOneById(messageId);
+
+	if (!message) {
+		throw new Error('error-invalid-message_id');
+	}
+
+	const user = await Users.findOneById(uid);
+
+	if (!user) {
+		throw new Error('error-invalid-user');
+	}
+
+	const { rid } = message;
+	// If the user can't access the room where the message is, report that the message id is invalid
+	const room = await Rooms.findOneById(rid);
+	if (!room || !(await canAccessRoomAsync(room, { _id: uid }))) {
+		throw new Error('error-invalid-message_id');
+	}
+
+	const reportedBy = {
+		_id: user._id,
+		username: user.username,
+		name: user.name,
+		active: user.active,
+		avatarETag: user.avatarETag,
+		createdAt: user.createdAt,
```

Comment:
Not including these props, does it mean:
```
const reportedBy = await Users.findOneById(uid);
// storing all the fields received
```

OR 
```
const reportedBy = await Users.findOneById(uid, {projection});
// projection will include the name, active, avatarEtag
```

Because, we need the above props, on the Moderation Console, to show the user details of the user who has reported the message

Reply:
yep you need this information to be displayed on the table.

it doesn't mean that you need to store those values. what happens if the user changes the name or the avatar in the middle? are you willing to keep all information updated? I think you can build this info on run time on the endpoint like we for for threads.. or you can use `useUserData` on the frontend that returns the values that you need

Reply:
the idea is to keep historical data as it was when it was reported..

I can see storing the "reporter" data is less important than who is being reported, which I think is just saved within the message reported right?

but anyways, this is why it is doing like this instead of fetching this data on frontend or in the endpoint before returning the data.